File: Universal Windows Platform Solution/Phantom of the West/Phantom of the West/Visual Novel/Chapters/Chapter 2/Chapter2.cs
using System.Collections.Generic;

namespace Phantom_of_the_West.Visual_Novel.Chapters.Chapter_2
{
	internal class Chapter2 : VoicedChapter
	{
		internal static Dictionary<int, StoryChoiceListDelegate> dictionary = new Dictionary<int, StoryChoiceListDelegate>
		{
			{2001, Ch2Title},
			{2002, Ch2TimeAgo},
			{2003, Ch2ThereWereFourGods},
			{2004, Ch2ThereWasSezja},
			{2005, Ch2ThereWasBaqer},
			{2006, Ch2ThereWasVaNal},
			{2007, Ch2ThereWasBekShtii},
			{2008, Ch2LectureEnded},
			{2009, Ch2IWentToProfArvin},
			{2010, Ch2IGreetProf},
			{2011, Ch2ProfGreetsMe},
			{2012, Ch2IConfideToProf},
			{2013, Ch2ProfDismissesMe}
		};

		internal Chapter2() : base(dictionary, 2003)
		{
		}

		private static StoryChoiceList Ch2Title()
		{
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { StoryChoice.FromResource("ch2Title", 2002) });
		}

		private static StoryChoiceList Ch2TimeAgo()
		{
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { StoryChoice.FromResource("ch2TimeAgo", 2003) });
		}

		private static StoryChoiceList Ch2ThereWereFourGods()
		{
			return StoryChoiceList.FromResource("Professor-Lecture.png", new IStoryChoice[] { StoryChoice.FromResource("ch2ThereWereFourGods", 2004) });
		}

		private static StoryChoiceList Ch2ThereWasSezja()
		{
			return StoryChoiceList.FromResource("Sezja-Intro.png", new IStoryChoice[] { StoryChoice.FromResource("ch2ThereWasSezja", 2005) });
		}

		private static StoryChoiceList Ch2ThereWasBaqer()
		{
			return StoryChoiceList.FromResource("Baqer-Intro.png", new IStoryChoice[] { StoryChoice.FromResource("ch2ThereWasBaqer", 2006) });
		}

		private static StoryChoiceList Ch2ThereWasVaNal()
		{
			return StoryChoiceList.FromResource("Va'Nal-Intro.png", new IStoryChoice[] { StoryChoice.FromResource("ch2ThereWasVaNal", 2007) });
		}

		private static StoryChoiceList Ch2ThereWasBekShtii()
		{
			return StoryChoiceList.FromResource("Bek'Shtii-Intro.png", new IStoryChoice[] { StoryChoice.FromResource("ch2ThereWasBekShtii", 2008) });
		}

		private static StoryChoiceList Ch2LectureEnded()
		{
			return StoryChoiceList.FromResource("Kaden-in-Lecture.png", new IStoryChoice[] { StoryChoice.FromResource("ch2LectureEnded", 2009) });
		}

		private static StoryChoiceList Ch2IWentToProfArvin()
		{
			return StoryChoiceList.FromResource("Kaden-Walks-in-School-Hall.png", new IStoryChoice[] { StoryChoice.FromResource("ch2IWentToProfArvin", 2010) });
		}

		private static StoryChoiceList Ch2IGreetProf()
		{
			return StoryChoiceList.FromResource("Kaden-Talks-to-Prof.png", new IStoryChoice[] { StoryChoice.FromResource("ch2IGreetProf", 2011) });
		}

		private static StoryChoiceList Ch2ProfGreetsMe()
		{
			return StoryChoiceList.FromResource("Kaden-Talks-to-Prof.png", new IStoryChoice[] { StoryChoice.FromResource("ch2ProfGreetsMe", 2012) });
		}

		private static StoryChoiceList Ch2IConfideToProf()
		{
			return StoryChoiceList.FromResource("Kaden-Talks-to-Prof.png", new IStoryChoice[] { StoryChoice.FromResource("ch2IConfideToProf", 2013) });
		}

		private static StoryChoiceList Ch2ProfDismissesMe()
		{
			return StoryChoiceList.FromResource("Kaden-Talks-to-Prof.png", new IStoryChoice[] { StoryChoice.FromResource("ch2ProfDismissesMe", 3001) });
		}
	}
}

File: Android Project/app/src/main/java/com/christopherraleigh/phantomofthewest/visual_novel/chapters/chapter_2/Chapter2.java
package com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_2;

import com.christopherraleigh.phantomofthewest.R;
import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoice;
import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoiceList;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.Chapter;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.StoryChoice;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.StoryChoiceList;

import java.util.Arrays;

/**
 * Created by christopherraleigh on 2016-05-11.
 */
public class Chapter2 extends Chapter {
    private static StoryChoiceList ch2Title = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2Title, 2002)}));

    private static StoryChoiceList ch2TimeAgo = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2TimeAgo, 2003)}));

    private static StoryChoiceList ch2ThereWereFourGods = new StoryChoiceList(R.drawable.professor_lecture,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2ThereWereFourGods, 2004)}));

    private static StoryChoiceList ch2ThereWasSezja = new StoryChoiceList(R.drawable.sezja_intro,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2ThereWasSezja, 2005)}));

    private static StoryChoiceList ch2ThereWasBaqer = new StoryChoiceList(R.drawable.baqer_intro,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2ThereWasBaqer, 2006)}));

    private static StoryChoiceList ch2ThereWasVaNal = new StoryChoiceList(R.drawable.vanal_intro,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2ThereWasVaNal, 2007)}));

    private static StoryChoiceList ch2ThereWasBekShtii = new StoryChoiceList(R.drawable.bekshtii_intro,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2ThereWasBekShtii, 2008)}));

    private static StoryChoiceList ch2LectureEnded = new StoryChoiceList(R.drawable.kaden_in_lecture,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2LectureEnded, 2009)}));

    private static StoryChoiceList ch2IWentToProfArvin = new StoryChoiceList(R.drawable.kaden_walks_in_school_hall,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2IWentToProfArvin, 2010)}));

    private static StoryChoiceList ch2IGreetProf = new StoryChoiceList(R.drawable.kaden_talks_to_prof,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2IGreetProf, 2011)}));

    private static StoryChoiceList ch2ProfGreetsMe = new StoryChoiceList(R.drawable.kaden_talks_to_prof,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2ProfGreetsMe, 2012)}));

    private static StoryChoiceList ch2IConfideToProf = new StoryChoiceList(R.drawable.kaden_talks_to_prof,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2IConfideToProf, 2013)}));

    private static StoryChoiceList ch2ProfDismissesMe = new StoryChoiceList(R.drawable.kaden_talks_to_prof,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch2ProfDismissesMe, 3001)}));

    @Override
    public IStoryChoiceList goToState(int id) {
        switch (id) {
            case 2001:
                return ch2Title;
            case 2002:
                return ch2TimeAgo;
            case 2003:
                return ch2ThereWereFourGods;
            case 2004:
                return ch2ThereWasSezja;
            case 2005:
                return ch2ThereWasBaqer;
            case 2006:
                return ch2ThereWasVaNal;
            case 2007:
                return ch2ThereWasBekShtii;
            case 2008:
                return ch2LectureEnded;
            case 2009:
                return ch2IWentToProfArvin;
            case 2010:
                return ch2IGreetProf;
            case 2011:
                return ch2ProfGreetsMe;
            case 2012:
                return ch2IConfideToProf;
            case 2013:
                return ch2ProfDismissesMe;
        }
        return null;
    }
}
